src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -252,14 +252,22 @@ template <typename InterfaceType> void RadioSpinel<InterfaceType>::ResetRcp(bool
     mIsReady    = false;
     mWaitingKey = SPINEL_PROP_LAST_STATUS;
 
-    if (aResetRadio && (SendReset(SPINEL_RESET_STACK) == OT_ERROR_NONE) && (WaitResponse(false) == OT_ERROR_NONE))
+    if (aResetRadio)
```

Comment:
If I am not mistaken we can change this line to `if (aResetRadio) && (SendReset(SPINEL_RESET_STACK) == OT_ERROR_NONE)` and keep the rest as in this PR?

This way, if `SendReset()` fails we fall down to try to do hw reset? 

Does this make sense?

Reply:
The root cause of the issue is that the RESET reason is missed (see my explain in the next comment). For the SPI interface, it won't miss the RESET reason from the RCP. Checking the response could help the host know whether the RCP is reset successfully.  So I think it is better to check the `WaitResponse` here.